tests/php/modules/shortcodes/test-class.tweet.php

```diff
@@ -169,4 +169,69 @@ public function test_shortcodes_tweet_card_via_oembed_http_request() {
 		$this->assertContains( '<blockquote class="twitter-tweet"', $shortcode_content );
 		// Not testing here for actual URL because wp_oembed_get might return a shortened Twitter URL with t.co domain
 	}
+
+	/**
+	 * Gets the test data for test_shortcodes_tweet_amp().
+	 *
+	 * @return array The test data.
+	 */
+	public function get_tweet_amp_data() {
+		$tweet_id       = 95234262;
+		$default_height = 480;
+		$default_width  = 600;
+
+		return array(
+			'no_attributes'         => array(
+				'[tweet]',
+				'<!-- Invalid tweet id -->',
+			),
+			'id_in_attributes'      => array(
+				'[tweet ' . $tweet_id . ']',
+				'<amp-twitter data-tweetid="'. $tweet_id .'" layout="responsive" width="' . $default_width . '" height="' . $default_height .'"></amp-twitter>',
+			),
+			'width_in_attributes'   => array(
+				'[tweet ' . $tweet_id . ' width=300]',
+				'<amp-twitter data-tweetid="'. $tweet_id .'" layout="responsive" width="300" height="' . $default_height .'"></amp-twitter>',
+			),
+			'0_width_in_attributes' => array(
+				'[tweet ' . $tweet_id . ' width=0]',
+				'<amp-twitter data-tweetid="'. $tweet_id .'" layout="responsive" width="' . $default_width . '" height="' . $default_height .'"></amp-twitter>',
+			),
+			'id_as_part_of_url'     => array(
+				'[tweet https://twitter.com/jetpack/status/' . $tweet_id . ']',
+				'<amp-twitter data-tweetid="'. $tweet_id .'" layout="responsive" width="' . $default_width . '" height="' . $default_height .'"></amp-twitter>',
+			),
+			'id_in_tweet_attribute' => array(
+				'[tweet tweet=' . $tweet_id . ']',
+				'<amp-twitter data-tweetid="'. $tweet_id .'" layout="responsive" width="' . $default_width . '" height="' . $default_height .'"></amp-twitter>',
+			),
+		);
+	}
+
+	/**
+	 * Test the AMP-compatible [tweet] shortcode on an AMP endpoint.
+	 *
+	 * @dataProvider get_tweet_amp_data
+	 * @since 8.0.0
+	 *
+	 * @param string $shortcode_content The shortcode, like [tweet 1234].
+	 * @param string $expected The expected return value of the function.
+	 */
+	public function test_shortcodes_tweet_amp( $shortcode_content, $expected ) {
+		add_filter( 'jetpack_is_amp_request', '__return_true' );
```

Comment:
Could you add the following?

```suggestion
		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
			self::markTestSkipped( 'WordPress.com does not run the latest version of the AMP plugin yet.' );
			return;
		}

		add_filter( 'jetpack_is_amp_request', '__return_true' );
```